src/core/datastore/repositories/network-repository.js

```diff
@@ -0,0 +1,100 @@
+import { KinveyRequest, RequestMethod, DeltaFetchRequest } from '../../request';
+
+import { Repository } from './repository';
+import { ensureArray } from '../../utils';
+import { buildCollectionUrl } from '../utils';
+
+/**
+ * @typedef RequestOptions
+ * @property {Operation} operation
+ * @property {Object} options
+ * @property {Query} [options.query]
+ * @property {Query} [options.query]
+ * @property {Object} [options.aggregation]
+ * @property {Number} [options.timeout]
+ * @property {Object} [options.properties]
+ * @property {Boolean} [options.skipBL]
+ * @property {Boolean} [options.trace]
+ */
+
+export class NetworkRepository extends Repository {
+  read(collection, query, options = {}) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, query, null, null, options);
+    return this._makeHttpRequest(requestConfig, options.useDeltaFetch);
+  }
+
+  readById(collection, entityId, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, null, entityId, null, options);
+    return this._makeHttpRequest(requestConfig);
+  }
+
+  create(collection, entities, options) {
+    return this._processBatch(collection, RequestMethod.POST, entities, null, options);
```

Comment:
This is a method internal to the NetworkRepository class. The API of repositories supports batch operations. The public API (this of data stores) does not allow for batch operations, so users of the SDK will not be able to do bulk operations.

The Repository API, however, is internal and I believe it's best to have it work with either a single entity, or an array of entities. The repository would best know how to handle bulk operations and this should not matter to anyone using a repository. If they would like to have sequential inserts (for instance), they can call `create()` multiple times. But if they want a bulk insert, they would call it with an array and let the repository handle this as best as it can. So the consideration of error handling is entirely up to us, but we keep the repository API flexible because different implementations can do transactions and other atomic operations.